adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala

```diff
@@ -122,16 +135,16 @@ object MdTag {
                     mdTagStringOffset += deletedBases.size
                   }
                 }
-                cigarReferencePosition += cigarElement.getLength
                 cigarIdx += 1
+                cigarOperatorIndex = 0
               }
               case _ => throw new IllegalArgumentException(s"CIGAR ${cigar.toString} indicates deletion found but no deleted bases in MDTag $mdTagInput")
             }
           }
           case _ if cigarElement.getOperator.consumesReferenceBases() => {
             referencePos += cigarElement.getLength
-            cigarReferencePosition += cigarElement.getLength
             cigarIdx += 1
+            cigarOperatorIndex = 0
           }
           case _ => {
             cigarIdx += 1
```

Comment:
is the `cigarOperatorIndex = 0` not necessary here?

mind commenting these last two cases a little bit? I assume the main interesting cigar operator being passed over here is `I`/insertion, which MDTags don't care about... any other cases being handled here worth noting?